src/librustc_lint/unused.rs

```diff
@@ -103,11 +103,33 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for UnusedResults {
                         cx.span_lint(UNUSED_MUST_USE, expr.span, msg);
                         op_warned = true;
                     },
-                    _ => {},
+                    hir::BiAdd | hir::BiSub | hir::BiDiv | hir::BiMul | hir::BiRem => {
+                        let msg = "unused arithmetic operation which must be used";
+                        cx.span_lint(UNUSED_MUST_USE, expr.span, msg);
+                        op_warned = true;
+                    },
+                    hir::BiAnd | hir::BiOr => {
+                        let msg = "unused logical operation which must be used";
+                        cx.span_lint(UNUSED_MUST_USE, expr.span, msg);
+                        op_warned = true;
+                    },
+                    hir::BiBitXor | hir::BiBitAnd | hir::BiBitOr | hir::BiShl | hir::BiShr => {
+                        let msg = "unused bitwise operation which must be used";
+                        cx.span_lint(UNUSED_MUST_USE, expr.span, msg);
+                        op_warned = true;
+                    },
+                }
+            }
+            if let hir::ExprUnary(un_op, ..) = expr.node {
```

Comment:
I would combine this and the `if let hir::ExprBinary` bit above into a single `match`.

Reply:
What is the best practice for combining these two match statements?  Combining them seems like it would require nested matches since `ExprBinary` and `ExprUnary` are different types.  I have to match on `op.node` for binary operators but only on `op` for unary operators.  Should I implement this nested system or is there a better way?

Reply:
A nested match is fine. It's mostly about eliminating the redundant code here.